input-output-hk/cardano-rosetta
# fix: Delay cardano-db-sync-startup until cardano-node is at network tip

# Description
Concurrently syncing `cardano-db-sync` with `cardano-node` from genesis can result in a non-recoverable state due this [known issue](https://github.com/input-output-hk/cardano-db-sync/issues/434). It can be mitigated by allowing the node to sync up to the network tip first, then starting `cardano-db-sync`.

# Proposed Solution
- Adds a small TypeScript program to manage the startup sequence
- The current protocol version is compared to the last known as
 per Shelley genesis.
- If in last known era the chain growth is sampled with a 9 second window
- Delta greater than 3 is still syncing
- To increase confidence of a low chain growth, a second sample is captured
after a short delay. This attempts to reduce the chance of a network pause,
or epoch boundary delay.
- removes basic bash script previously delaying by 5 seconds
# Important Changes Introduced
- removes basic bash script previously delaying by 5 seconds

# Testing
- `docker run  -e LOGGER_MIN_SEVERITY=debug ...`
- Start up a fresh instance on the mainnet, let it sync, and ensure `cardano-db-sync` is started once `cardano-node` is near the network tip.
- Restart the instance, ensure it starts after the 23 second delay
